specs/agents/sanitization.md

```diff
@@ -16,7 +16,7 @@ how an agent will sanitize data.
 |                |   |
 |----------------|---|
 | Type           | `List<`[`WildcardMatcher`](../../tests/agents/json-specs/wildcard_matcher_tests.json)`>` |
-| Default        | `password, passwd, pwd, secret, *key, *token*, *session*, *credit*, *card*, authorization, set-cookie` |
+| Default        | `password, passwd, pwd, secret, *key, *token*, *session*, *credit*, *card*, *auth*, set-cookie` |
```

Comment:
I started the [Node.js implementation](https://github.com/elastic/apm-agent-nodejs/pull/2495) and ran into a test failure that I think should be discussed:  The `*auth*` pattern results in redacting the HTTP/2 `:authority` pseudo-header (https://datatracker.ietf.org/doc/html/rfc7540#section-8.1.2.3) -- the moral equivalent of the HTTP/1.1 `Host` header. 

I think we likely don't want to redact this, but I don't know that it causes any harm.  The `WildcardMatcher` format doesn't support exclusions, so there isn't a way to support this directly in the value of `sanitize_field_names`. I the "Sanitizing Values" section below could be specified to exclude the four spec'd HTTP/2 pseudo-headers.

Reply:
Should we even capture pseudo-headers? Aren't they redundant, if we're capturing the URL and method?

Reply:
> Should we even capture pseudo-headers? 

I would happily defer to those with more experience with HTTP/2 than I.

Given language like the following from RFC 7540, I can concoct a theoretical example where having the `:authority` header value in trace data could be helpful for debugging a proxy. However, that may be a stretch.

```
      An
      intermediary that converts an HTTP/2 request to HTTP/1.1 MUST
      create a Host header field if one is not present in a request by
      copying the value of the ":authority" pseudo-header field.
```


> Aren't they redundant, if we're capturing the URL and method?

Yes, they are (modulo a Node.js APM agent bug that isn't properly gathering `transaction.context.request.url.*` fields for HTTP/2).